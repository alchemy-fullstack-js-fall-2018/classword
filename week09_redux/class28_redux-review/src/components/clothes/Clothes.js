import React from 'react';
import PropTypes from 'prop-types';
import styles from './Clothes.css';

const Clothes = ({ clothes, totalClothes }) => {
  const clothesItems = clothes.map(c => {
    return <li key={c.id}>{c.id} - {c.type} - {c.color}</li>;
  });
  return (
    <div className={styles.clothes}>
      <h3>{totalClothes}</h3>
      <ul>
        {clothesItems}
      </ul>
    </div>
  );
};

Clothes.propTypes = {
  clothes: PropTypes.array.isRequired,
  totalClothes: PropTypes.number.isRequired
};

export default Clothes;
