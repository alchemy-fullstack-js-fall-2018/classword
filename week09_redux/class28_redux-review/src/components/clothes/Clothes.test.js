import React from 'react';
import { shallow } from 'enzyme';
import Clothes from './Clothes';

describe('Clothes component', () => {
  const clothes = [
    { type: 'shirt', color: 'red', id: 'abcd1234' }
  ];
  it('matches a snapshot', () => {
    expect(true);
    const wrapper = shallow(<Clothes
      clothes={clothes}
      totalClothes={10} />);
    expect(wrapper).toMatchSnapshot();
  });
});
