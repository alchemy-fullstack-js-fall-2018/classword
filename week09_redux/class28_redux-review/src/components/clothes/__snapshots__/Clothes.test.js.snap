// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clothes component matches a snapshot 1`] = `
<div
  className="clothes"
>
  <h3>
    10
  </h3>
  <ul>
    <li
      key="abcd1234"
    >
      abcd1234
       - 
      shirt
       - 
      red
    </li>
  </ul>
</div>
`;
